src/System.Private.CoreLib/src/System/Reflection/Emit/DynamicILGenerator.cs

```diff
@@ -901,8 +913,109 @@ internal byte[] LocalSignature
         public DynamicMethod DynamicMethod { get { return m_method; } }
         internal DynamicScope DynamicScope { get { return m_scope; } }
 
+        public void SetCode(byte[] code, int maxStackSize)
+        {
+            m_code = (code != null) ? (byte[])code.Clone() : Array.Empty<byte>();
+            m_maxStackSize = maxStackSize;
+        }
+
+        [CLSCompliant(false)]
+        public unsafe void SetCode(byte* code, int codeSize, int maxStackSize)
+        {
+            if (codeSize < 0)
+                throw new ArgumentOutOfRangeException(nameof(codeSize), SR.ArgumentOutOfRange_GenericPositive);
+            if (codeSize > 0 && code == null)
+                throw new ArgumentNullException(nameof(code));
+
+            m_code = new byte[codeSize];
+            for (int i = 0; i < codeSize; i++)
```

Comment:
nit use `.Length` to elide the bounds check
```
for (int i = 0; i < m_code.Length; i++)
```
or better  use local var to be sure
```csharp
byte[] codes = new byte[codeSize];
for (int i = 0; i < codes.Length; i++)
{
    codes[i] = *code;
    code++;
}

m_code = codes;
```

Reply:
Couldn't we just use span?
```C#
m_code = new Span<byte>(code, codeSize).ToArray();
```